Source/WebCore/animation/AnimationTimelinesController.cpp

```diff
@@ -107,11 +112,11 @@ void DocumentTimelinesController::updateAnimationsAndSendEvents(ReducedResolutio
     m_frameRateAligner.beginUpdate(timestamp, previousTimelineFrameRate);
 
     // 1. Update the current time of all timelines associated with document passing now as the timestamp.
-    Vector<Ref<DocumentTimeline>> timelinesToUpdate;
+    Vector<Ref<AnimationTimeline>> timelinesToUpdate;
     Vector<Ref<WebAnimation>> animationsToRemove;
     Vector<Ref<CSSTransition>> completedTransitions;
     for (auto& timeline : protectedTimelines) {
-        auto shouldUpdateAnimationsAndSendEvents = timeline->documentWillUpdateAnimationsAndSendEvents();
+        auto shouldUpdateAnimationsAndSendEvents = is<DocumentTimeline>(timeline) ? dynamicDowncast<DocumentTimeline>(timeline)->documentWillUpdateAnimationsAndSendEvents() : DocumentTimeline::ShouldUpdateAnimationsAndSendEvents::No;
```

Comment:
The fallback value here should be `Yes` actually, at least until we determine that other timeline types should have a similar function that determines whether they need an update (possibly if we determine that scroll offset hasn't changed since the last page update).

Reply:
Another thing we could do is to make that function not specific to `DocumentTimeline` and have the `AnimationTimeline` superclass just return `Yes` and let `DocumentTimeline` have its own override that does what it already does. That's probably a better thing to do. Can be done as a separate patch.

Reply:
Yeah I was planning on making a separate patch to add the `ScrollTimeline` and `ViewTimeline` objects to the `AnimationTimelinesController` and add the virtual functions on `AnimationTimeline`/implement the relevant ones on `ScrollTimeline` and `ViewTimeline`.